src/components/Shared/SendModal/SendLN/SendLN.tsx

```diff
@@ -1,88 +1,59 @@
+
+import { useForm } from "react-hook-form";
+import type { SubmitHandler } from "react-hook-form";
 
 const SendLn: FC<SendLnProps> = (props) => {
   const appCtx = useContext(AppContext);
   const { balance, onChangeInvoice, onConfirm } = props;
   const { t } = useTranslation();
 
-  const invoiceInput = useRef<HTMLInputElement>(null);
-  const [isFormValid, setIsFormValid] = useState(true);
-  const [errorMessage, setErrorMessage] = useState("");
-
-  const validateInput = (validity: HTMLInputElement["validity"]) => {
-    setIsFormValid(validity.valid);
-    setErrorMessage("");
-
-    if (validity.valueMissing) {
-      setErrorMessage(t("forms.validation.lnInvoice.required"));
-      return false;
-    }
-
-    if (validity.patternMismatch) {
-      setErrorMessage(t("forms.validation.lnInvoice.patternMismatch"));
-      return false;
-    }
+  interface IFormInputs {
+    invoiceInput: string;
+  }
 
-    return true;
-  };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid, submitCount },
+  } = useForm<IFormInputs>({
+    mode: "onChange",
+  });
 
-  const onChange = (event: ChangeEvent<HTMLInputElement>) => {
-    validateInput(event.target.validity);
-    onChangeInvoice(event);
-  };
-
-  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const validity = invoiceInput?.current?.validity;
-
-    if (validity) {
-      const isFormValid = validateInput(validity);
-
-      if (isFormValid) {
-        onConfirm(event);
-      }
-    }
-  };
+  const onSubmit: SubmitHandler<IFormInputs> = (_data) => onConfirm();
 
   return (
-    <form onSubmit={onSubmit} noValidate>
+    <form onSubmit={handleSubmit(onSubmit)}>
       <h3 className="text-xl font-bold">{t("wallet.send_lightning")}</h3>
 
       <p className="my-5">
         <span className="font-bold">{t("wallet.balance")}:&nbsp;</span>
         {balance} {appCtx.unit}
       </p>
 
-      <label className="label-underline" htmlFor="invoiceInput">
-        {t("wallet.invoice")}
-      </label>
-
-      <input
-        id="invoiceInput"
-        ref={invoiceInput}
-        type="text"
-        onChange={onChange}
-        required
-        pattern="(lnbc|lntb)\w+"
-        className={isFormValid ? "input-underline" : "input-error"}
-        placeholder="lnbc..."
+      <InputField
+        {...register("invoiceInput", {
+          required: t("forms.validation.lnInvoice.required") as string,
+          pattern: {
+            value: /^(lnbc|lntb)\w+/i,
+            message: t("forms.validation.lnInvoice.patternMismatch"),
+          },
+          onChange: onChangeInvoice,
+        })}
+        label={t("wallet.invoice")}
+        errorMessage={errors.invoiceInput}
       />
 
-      <p className="text-red-500">{errorMessage}</p>
-
       <button
         type="submit"
         className="bd-button p-3 my-3"
-        disabled={!isFormValid}
+        disabled={submitCount > 0 && !isValid}
```

Comment:
Here the available balance should be taken into account as well, so I can't send more than available